nemo/collections/asr/modules/conformer_encoder.py

```diff
@@ -240,16 +258,7 @@ def forward(self, audio_signal, length=None):
         audio_signal = torch.transpose(audio_signal, 1, 2)
         return audio_signal, length
 
-    @staticmethod
-    def make_pad_mask(seq_lens, max_time, device=None):
+    def make_pad_mask(self, audio_size, seq_lens):
```

Comment:
Refer to response in thread. Why is this being made static ? For export ? This is problematic because I'm using this in a subsequent PR for masked SE block in Citrinets and it can't be made static there.

Reply:
Why, it's the other way around - I removed static.

Reply:
Now the mask size is limited at init time - previously it was dependent on input sequence length and didn't depend on some notion of max seq length at train time that had to be set. 

For example it is very common to take pretrained Conformer trained on 16s max length and infer on 2 mins of audio in offline. Same for Citrinet, train on 16 s eval on 2 hours audio. Unlike positional embeddings, these masks don't need to be learned either so it makes more sense to construct them at runtime. 

Basically, Init should not be used to define max length,but should adjust to current provided max length.

Reply:
It looks to me that this function is not static in sense of the maximum audio lengths. self.seq_range is fixed but it is dependent on the max_len=5000 which is a large number already. I would add a method to give the user the ability to extend it after training is done.

Reply:
Yes, I have made that functionality available in set_pos_emb_max_len. So it's called in init() for config value and can also be called explicitly to increase max len for inference. For regular pytorch inference, I can also provide separate forward() and forward_for_export() so that forward() still has dynamic length - would you like that, too ?
Mask can't be fully dynamic with TRT, unfortunately - that does raise a requirement for possible 'config adjustment' during export() - for RIVA, we can handle that with direct config modification though.

Reply:
Right for pytorch only inference and training let's make it dynamic and for export force fixed mask

Reply:
Done.